pkg/controller/operators/olm/requirements.go

```diff
@@ -39,24 +40,41 @@ func (a *Operator) requirementStatus(strategyDetailsDeployment *install.Strategy
 			status.Status = v1alpha1.RequirementStatusReasonPresent
 			status.UUID = string(crd.GetUID())
 			statuses = append(statuses, status)
-			continue
+		} else {
+			served := false
+			for _, version := range crd.Spec.Versions {
+				if version.Name == r.Version {
+					if version.Served {
+						status.Status = v1alpha1.RequirementStatusReasonPresent
+						status.UUID = string(crd.GetUID())
+						statuses = append(statuses, status)
+						served = true
+					}
+					break
+				}
+			}
+
+			if !served {
+				status.Status = v1alpha1.RequirementStatusReasonNotPresent
+				met = false
+				statuses = append(statuses, status)
+				continue
+			}
 		}
 
-		served := false
-		for _, version := range crd.Spec.Versions {
-			if version.Name == r.Version {
-				if version.Served {
-					status.Status = v1alpha1.RequirementStatusReasonPresent
-					status.UUID = string(crd.GetUID())
-					statuses = append(statuses, status)
-					served = true
+		// Check if CRD has successfully registered with k8s API
```

Comment:
I suggest moving this code block (and the !registered) to above so that only bad statuses are written. Otherwise, you end up getting a "Present" status and a "PresetNotAvailable", which seems slightly confusing to me.